### PR TITLE
Fixed corner-case bug in CanonicalHyperCubeSet.

### PR DESCRIPTION
**Pull request by tanyaveksler**
Also added unit test for this case.

### CODE REVIEW

File: tests/classes_unit_tests/testCanonicalHyperCubeSetNew.py [hunk 1]
```diff
@@ -971,6 +973,33 @@ def test_contained_in_new(self):
...
 
+    def test_bug_in_contained(self):
+        BasePeerSet.reset()
+        BasePeerSet().add_peer("A")
+        BasePeerSet().add_peer("B")
+        BasePeerSet().add_peer("C")
+        ipv4_block = IpBlock.get_all_ips_block(True)
+        my_dimensions1 = ["src_peers", "dst_peers"]
+        my_dimensions2 = ["src_peers", "dst_peers", "protocols"]
+        conns1 = CanonicalHyperCubeSet.create_from_cube(my_dimensions1,
```
Comment (adisos) [inline tests/classes_unit_tests/testCanonicalHyperCubeSetNew.py:984]: why is `my_dimensions1` the first arg (all_dims) to `create_from_cube`, and not `my_dimensions2`?

Comment (tanyaveksler) [reply tests/classes_unit_tests/testCanonicalHyperCubeSetNew.py:984]: My mistake. Changed to my_dimensions2.

File: tests/classes_unit_tests/testCanonicalHyperCubeSetNew.py [hunk 2]
```diff
@@ -971,6 +973,33 @@ def test_contained_in_new(self):
...
+                                                        [BasePeerSet().get_peer_interval_of({"B"}),
+                                                         BasePeerSet().get_peer_interval_of({"A", "B", "C", ipv4_block})],
```
[the remark this answers is not in the document]
Comment (tanyaveksler) [reply tests/classes_unit_tests/testCanonicalHyperCubeSetNew.py:986]: Right (it appeared because I generated this example from the real example where `ipv4_block` was present).
Removed it from this example.

**Review by adisos — APPROVED** [review]
